core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -195,6 +195,16 @@ default ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connec
         return emptyList();
     }
 
+    /**
+     * Gets all of the columns on the specified table, or an empty list if the columns cannot be enumerated.
```

Comment:
I don't get this `or`. Is it designed for metadata queries or analyzer? If the latter we fail in case of error.